### PR TITLE
[Network] Fix'az network dns record-set txt add-record' is not idempotent

### PR DESCRIPTION
**Pull request by 00Kai0**
Closed https://github.com/Azure/azure-cli/issues/10467
--
**Description**

Add filters for recordsets to prevent duplicate records

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -2380,7 +2380,7 @@ def update_dns_soa_record(cmd, resource_group_name, zone_name, host=None, email=
 def add_dns_srv_record(cmd, resource_group_name, zone_name, record_set_name, priority, weight,
                        port, target, if_none_match=None):
     SrvRecord = cmd.get_models('SrvRecord', resource_type=ResourceType.MGMT_NETWORK_DNS)
-    record = SrvRecord(priority=priority, weight=weight, port=port, target=target)
+    record = SrvRecord(priority=int(priority), weight=int(weight), port=int(port), target=target)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:2383]: Arguments should add `type=int` instead

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 2]
```diff
@@ -661,10 +661,18 @@ def load_arguments(self, _):
...
+                c.argument('weight', type=int)
+                c.argument('port', type=int)
 
         with self.argument_context('network dns record-set {} remove-record'.format(item)) as c:
             c.argument('record_set_name', options_list=['--record-set-name', '-n'], help='The name of the record set relative to the zone.')
             c.argument('keep_empty_record_set', action='store_true', help='Keep the empty record set if the last record is removed.')
+            if item == 'srv':
+                c.argument('priority', type=int)
+                c.argument('weight', type=int)
+                c.argument('port', type=int)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:675]: You should update on the following existing lines https://github.com/Azure/azure-cli/blob/4bed2d2778f974b432ea5f2f1cd470d1897cfa7e/src/azure-cli/azure/cli/command_modules/network/_params.py#L725-L729

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**